src/test/java/co/org/osso/api/CalculatorControllerTests.java

```diff
@@ -0,0 +1,39 @@
+package co.org.osso.api;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CalculatorControllerTests {
+
+    TestLogger logger = new TestLogger();
+    CalculatorController target = new CalculatorController(new Calculator(logger), logger);
+
+    @Test
+    public void logFibonacciNumber(){
+        target.getFibonacci(10);
+        Assert.assertEquals(1, logger.logs.size());
+        Assert.assertEquals("/fibonacci/10", logger.logs.get(0));
+    }
+
+    @Test
+    public void logFactorialNumber(){
+        target.getFactorial(4);
+        Assert.assertEquals(1, logger.logs.size());
+        Assert.assertEquals("/factorial/4", logger.logs.get(0));
+    }
+}
+
+@Configuration
```

Comment:
This passes the build - but just barely so.

If you try and run the application, it will fail to start up. The build passes because all tests can correctly build the spring context, but this isn't possible by just running the app because it doesn't know which logger to give the CalculatorController. I think our problems are solved if we delete this annotation and put `@Component` on the `DefaultLogger`. `Configuration` is a special type of `Component` that can also define new `@Bean`s as methods, but we're not using that here.